src/Collection.php

```diff
@@ -221,8 +222,9 @@ public function __toString()
      * @throws InvalidArgumentException for parameter/option parsing errors
      * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
      */
-    public function aggregate(array $pipeline, array $options = [])
+    public function aggregate(array|Pipeline $pipeline, array $options = [])
```

Comment:
I don't know how I missed this while we were planning things, but this is a BC break, as any class extending `Collection` and overriding `aggregate` will be incompatible due to a narrower parameter type. I don't know if there's a way to solve this in 1.x, as I'd hate to defer this feature to 2.x.

Reply:
The compromise we can make for the moment is to accept an array of stages.

```php
$pipeline = new Pipeline(...);
$results = $collection->aggregate(iterator_to_array($pipeline));
```